lib/jwt_signed_request.rb

```diff
@@ -28,16 +32,20 @@ def self.sign(method:, path:, body: EMPTY_BODY, headers:, secret_key:, algorithm
     )
   end
 
-  def self.verify(request:, secret_key:, algorithm: nil)
+  def self.verify(request:, secret_key:, algorithm: nil, leeway: 0)
+    # algorithm is deprecated and will be removed in future
+    verify = true
     jwt_token = Headers.fetch('Authorization', request)
-    algorithm ||= DEFAULT_ALGORITHM
 
     if jwt_token.nil?
       raise MissingAuthorizationHeaderError, "Missing Authorization header in the request"
     end
 
     begin
-      claims = JWT.decode(jwt_token, secret_key, algorithm)[0]
+      #TODO Once JWT v2.0.0 has been released, we should upgrade to it and start using `exp_leeway` instead
+      # 'leeway' will still work, but 'exp_leeway' is more explicit and is the documented way to do it.
+      # see https://github.com/jwt/ruby-jwt/pull/187
+      claims = JWT.decode(jwt_token, secret_key, verify, {leeway: leeway.to_i})[0]
```

Comment:
doesn't `exp_leeway` already exist?

https://github.com/jwt/ruby-jwt/blob/master/lib/jwt/verify.rb#L72-L74
Im ok to use the global leeway for now, as we don't use `iat` or `nbf` so it doesn't affect it much.